### PR TITLE
Scope support

### PR DESCRIPTION
**Pull request by brendt**
This PR is based on `model-support` for review. #1 must be merged before this one can target master.

### CODE REVIEW

**Comment by Gummibeer** [comment]
After [spatie/enum:2.1.0](https://github.com/spatie/enum/releases/tag/2.1.0) we don't need the mapping of values in the scopes anymore. Instead the scopes should also respect the possible cast to integer - introduced in https://github.com/spatie/laravel-enum/commit/61e93c93b89623e4cb02247fbb594f7c92c4a30d .
I would also like to have a `whereEnum($column, $value)` and a second `whereInEnum($column, $values)` - same for `not`. I think that it should be possible to have a scope which would fail if multiple values are passed. So far I know the default laravel where methods also don't auto-switch depending on the argument type(?).

File: src/HasEnums.php [hunk 1]
```diff
@@ -89,9 +182,39 @@ protected function getEnumClass(string $key): string
...
         );
     }
+
+    /**
+     * @param Builder $builder
+     * @param string $method
+     * @param string $key
+     * @param int|string|Enumerable|int[]|string[]|Enumerable[] $enumerables
+     */
+    protected function buildEnumScope(
```
Comment (Gummibeer) [inline src/HasEnums.php]: In https://github.com/Astrotomic/laravel-translatable we have a similar approach to reduce the code duplicates and unify the logic.

File: src/HasEnums.php [hunk 2]
```diff
@@ -28,6 +30,86 @@ public function getAttribute($key)
             : $value;
     }
 
+    /**
```
Comment (Gummibeer) [inline src/HasEnums.php]: These doc-tags are only for IDEs - and the `@see` annotation for faster debugging.

**Comment by Gummibeer** [comment]
@brendt I've opened this PR because it's now on `master` level and I've also added the missing `or` scopes.

**Comment by Gummibeer** [comment]
@freekmurze could you check this PR? @brendt and I have written the code.

File: src/HasEnums.php [hunk 3]
```diff
@@ -28,6 +30,86 @@ public function getAttribute($key)
...
+     * @param Builder $builder
```
Comment (freekmurze) [inline src/HasEnums.php]: We usually use fqcn in docblocks

Comment (Gummibeer) [reply src/HasEnums.php]: https://github.com/spatie/laravel-enum/pull/4/commits/4aa4a0b6e341290b295db81dbe47807b8b346465

**Pull request merged**